### PR TITLE
Add lib wrapper for UUID validation.

### PR DESCRIPTION
**Pull request by nikolaevas**
### Description (*)
Fix for the async-import project. Move third-party library dependency to core repository.

### Fixed Issues (if relevant)
1. https://github.com/magento/async-import/issues/148 Add lib wrapper for UUID generation/validation #148

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: lib/internal/Magento/Framework/DataObject/IdentityValidator.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Framework\DataObject;
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/DataObject/IdentityValidator.php:8]: 1. Please add strict types for this file

File: lib/internal/Magento/Framework/DataObject/IdentityValidatorInterface.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+interface IdentityValidatorInterface
+{
+    /**
+     * Checks if uuid is valid
+     *
+     * @param string $value
+     *
+     * @return string
+     */
+    public function isValid($value);
```
Comment (ihor-sviziev) [inline lib/internal/Magento/Framework/DataObject/IdentityValidatorInterface.php]: Please add argument and return types there and in implementation

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
Hi @nikolaevas,
Your changes looks good.
Would be great to cover your changes with some tests. Will you be able to do that?

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6184](https://jira.corp.magento.com/browse/ENGCOM-6184) has been created to process this Pull Request
:eight_spoked_asterisk: @ihor-sviziev, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by slavvka** [comment]
The PR is currently under architecture review

**Comment by slavvka** [comment]
Hey @nikolaevas There is no dependency on Ramsey component in the framework's composer.json. Please add it.

**Comment by nuzil** [comment]
@slavvka done

**Comment by slavvka** [comment]
Hey @nuzil could you please fix the static tests?

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-6184](https://jira.corp.magento.com/browse/ENGCOM-6184) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Comment by slavvka** [comment]
BIC was approved https://jira.corp.magento.com/browse/MC-23065

**Pull request merged**